dev/update_website.py

```diff
@@ -0,0 +1,153 @@
+
+(How to run)
+python update_website <reef_home> <reef_version> <sha512> <release_notes_link>
+
+You can also see how to run the script with "python update_website.py -h"
+
+(Example)
+python update_website ~/reef 0.14.0 8f542ae...c4fed7 https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12315820&version=12333768
+"""
+
+
+import os
+import re
+import sys
+import argparse
+import datetime
+
+"""
+doap.rdf
+"""
+def update_doap(file, new_version):
+    changed_str = ""
+    after_release_tag = False
+
+    f = open(file, 'r')
+
+    # keep the part of the file before <release> tag as is
+    # necessary because name and created tags appear several times in the file
+    while True:
+        line = f.readline()
+        if not line:
+            break
+        if "<release>" in line:
+            after_release_tag = True
+        # update <name>, <revision> and <created> tags
+        if (("<name>" in line) or ("<revision>" in line)) and after_release_tag:
+            r = re.compile('>(.*?)<')
+            m = r.search(line)
+            old_version = m.group(1)
+            line = line.replace(old_version, new_version)
+        if "<created>" in line and after_release_tag:
+            r = re.compile('<created>(.*?)</created>')
+            m = r.search(line)
+            old_date = m.group(1)
+            new_date = datetime.date.today().strftime('%Y-%m-%d')
+            line = line.replace(old_date, new_date)
+        changed_str += line
+
+    f = open(file, 'w')
+    f.write(changed_str)
+    f.close()
+    print file
+
+
+"""
+downloads.md: replace all release information with new one
+"""
+def update_downloads(file, new_version, sha512, notes_link):
+    changed_str = ""
+    old_version = ""
+    f = open(file, 'r')
+    while True:
+        line = f.readline()
+        if not line:
+            break
+        # figure out old version
+        if "selected" in line:
+            r = re.compile('>(.*?)<')
+            m = r.search(line)
+            old_version = m.group(1)
+        if old_version != "" and old_version in line:
+            line = line.replace(old_version, new_version)
+        if "selected" in line:
+            # write this line and construct new one
+            changed_str += line
+            line = '    <option value="' + old_version + '">' + old_version + '</option>\n'
+        if "sha512Text" in line:
+            r = re.compile('<span id="sha512Text">(.*?)</span>')
+            m = r.search(line)
+            line = line.replace(m.group(1), sha512)
+        if "releaseNotesLink" in line:
+            r = re.compile('href="(.*?)"')
+            m = r.search(line)
+            line = line.replace(m.group(1), notes_link)
+        if "dotnetApiLink" in line:
+            r = re.compile('<span id="dotnetApiLink">(.*?)</span>')
+            m = r.search(line)
+            line = line.replace(m.group(1), '<a href="apidoc_net/' + new_version + '/index.html">.NET API</a>')
```

Comment:
If we know that the .NET apidocs link goes like 'apidoc_net/version/index.html', then we could just include 'apidoc_net/' and '/index.html' in the regex string, like the other `if` cases. Any reason not to? I'm assuming it's because we don't have that link in the downloads.md file yet?